Apps/Src/ReadSwitches.c

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+
+// Includes
+#include "ReadSwitches.h"
+
+// Macros
+#define OS_DELAY_MS(ms) \
+    OSTimeDlyHMSM(0, 0, 0, (ms), OS_OPT_TIME_HMSM_NON_STRICT, &err)
+#define MAX_SWITCH_LAG          100  // delay (ms) for checking switches while precharging
+#define PRECHARGE_DELAY_CYCLES  (PRECHARGE_ARRAY_DELAY*1000 / MAX_SWITCH_LAG)
+
+// Globals
+switch_states_t CurrSwitches;
```

Comment:
A bitmap that serves this same purpose exists in the BSP layer. Instead of using this to store the current switch states, use Switch_Read to access the bitmap.

Reply:
This was written before we had that bitmap; I've updated it to use Switches_Read 😄👍